app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -76,9 +77,13 @@ constructor(
 
 data class EventsOverviewUiState(
     val eventList: EventList = EventList(emptyList(), emptyList(), null),
+    var searchQuery: String = "",
+    var isFilterDialogOpen: Boolean = false,
+    var radiusInputFilter: Double = -1.0,
+    var freeEventsFilter: Boolean = false,
+    var categorySelectionFilter: List<EventCategory> = emptyList(),
     var viewList: Boolean = true,
     var tab: Tab = Tab.BROWSE,
-    val searchQuery: String = "",
 )
```

Comment:
Another reason to prohibit the creation of states using `EventsOverviewUiState()` directly. All of these should be `val` instead of `var`. Even if you are using `val`s, you will still be able to change the state from the viewmodel if instantiated correctly, here's an example:

```kotlin
  private val _state = MutableStateFlow(MyComponentState())
  val state = _state.asStateFlow()
```

The above snippet is located in a view model. You can then create said view model using

```kotlin
val vm = viewModel<MyComponentViewModel>()
```

and extract the state **as a flow** using

```kotlin
val state = vm.state.collectAsState() // or colectAsStateWithLifecycle()
```

Reply:
Resolved in PR #190